src/sage/schemes/curves/affine_curve.py

```diff
@@ -2321,16 +2405,35 @@ def _coordinate_functions(self):
         """
         return self._nonsingular_model[1].im_gens()
 
+    @lazy_attribute
+    def _pull_from_function_field(self):
```

Comment:
This is not my area, but it seems odd that `pull_from_function_field` returns an element of the fraction field, while `_pull_from_function_field` returns a map from the function field.

Reply:
I agree. I also changed `_lift_to_function_field` to `_map_to_function_field` since this returns a map too.